controller/StreamMetadataTasksTest.java

```diff
@@ -0,0 +1,2500 @@
+    private EventStreamWriterMock<CommitEvent> commitWriter;
+    private EventStreamWriterMock<AbortEvent> abortWriter;
+    @Mock
+    private KVTableMetadataStore kvtStore;
+    @Mock
+    private TableMetadataTasks kvtMetadataTasks;
+
+    @Before
+    public void setup() throws Exception {
+        zkServer = new TestingServerStarter().start();
+        zkServer.start();
+        zkClient = CuratorFrameworkFactory.newClient(zkServer.getConnectString(),
+                new ExponentialBackoffRetry(200, 10, 5000));
+        zkClient.start();
+        StreamMetrics.initialize();
+        TransactionMetrics.initialize();
+
+        StreamMetadataStore streamStore = getStore();
+        streamStorePartialMock = spy(streamStore); //create a partial mock.
+        ImmutableMap<BucketStore.ServiceType, Integer> map = ImmutableMap.of(BucketStore.ServiceType.RetentionService, 1,
+                BucketStore.ServiceType.WatermarkingService, 1);
+
+        bucketStore = StreamStoreFactory.createInMemoryBucketStore(map);
+        
+        TaskMetadataStore taskMetadataStore = TaskStoreFactory.createZKStore(zkClient, executor);
+        SegmentHelper segmentHelperMock = SegmentHelperMock.getSegmentHelperMock();
+        connectionFactory = new SocketConnectionFactoryImpl(ClientConfig.builder().build());
+        EventHelper helper = EventHelperMock.getEventHelperMock(executor, "host", ((AbstractStreamMetadataStore) streamStore).getHostTaskIndex());
+        streamMetadataTasks = spy(new StreamMetadataTasks(streamStorePartialMock, bucketStore, taskMetadataStore, segmentHelperMock,
+                executor, "host", new GrpcAuthHelper(authEnabled, "key", 300), requestTracker, helper));
+
+        streamTransactionMetadataTasks = new StreamTransactionMetadataTasks(
+                streamStorePartialMock, segmentHelperMock, executor, "host", 
+                new GrpcAuthHelper(authEnabled, "key", 300));
+
+        this.streamRequestHandler = new StreamRequestHandler(new AutoScaleTask(streamMetadataTasks, streamStorePartialMock, executor),
+                new ScaleOperationTask(streamMetadataTasks, streamStorePartialMock, executor),
+                new UpdateStreamTask(streamMetadataTasks, streamStorePartialMock, bucketStore, executor),
+                new SealStreamTask(streamMetadataTasks, streamTransactionMetadataTasks, streamStorePartialMock, executor),
+                new DeleteStreamTask(streamMetadataTasks, streamStorePartialMock, bucketStore, executor),
+                new TruncateStreamTask(streamMetadataTasks, streamStorePartialMock, executor),
+                streamStorePartialMock,
+                executor);
+        consumer = new ControllerService(kvtStore, kvtMetadataTasks, streamStorePartialMock, bucketStore, streamMetadataTasks,
+                streamTransactionMetadataTasks, segmentHelperMock, executor, null);
+        commitWriter = new EventStreamWriterMock<>();
```

Comment:
You are not cleaning this up in teardown. Please go through this class and cleanup any resources you allocate (i.e., anything that has a `shutdown` or `close` method).